requirements-dev.txt

```diff
@@ -19,7 +19,7 @@ reno>=3.4.0
 Sphinx>=3.0.0
 qiskit-sphinx-theme>=1.6
 qiskit-toqm>=0.0.4;platform_machine != 'aarch64' or platform_system != 'Linux'
-sphinx-autodoc-typehints<1.14.0
+sphinx-autodoc-typehints~=1.18
```

Comment:
This requirement is actually just `>=1,<2` - I don't know if you were intending to do something else (like `>=1.18`, perhaps?).